File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head==null || head.next==null || k<=0){
            return head;
        }
       ListNode last = head;
       int count = 1;
       while(last.next!=null){
        last = last.next;
        count++;
       }
       last.next = head;
       int rotations = k%count;
       int skip = count - rotations;
       ListNode newLast = head;
       for(int i=0;i<skip-1;i++){
        newLast = newLast.next;
       }
       head = newLast.next;
       newLast.next = null;
       return head;
    }
}

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if head==None or k==0 or head.next==None:
            return head
        count=1
        tail = head
        while tail.next:
            tail = tail.next
            count+=1
        k = k%count
        if k==0:
            return head
        moves = count-k
        temp = head
        for i in range (0,moves-1,1):
            temp = temp.next
        new_head = temp.next
        temp.next=None
        tail.next=head
        return new_head